include/hexahedron_element.tcc

```diff
@@ -563,4 +563,48 @@ inline Eigen::MatrixXi
   //clang-format on
   return indices;
 }
+
+//! Return indices of a face of the element
+//! 8-noded hexahedron
+template <>
+inline Eigen::MatrixXi
+    mpm::HexahedronElement<3, 8>::face_indices(unsigned face_id) const {
+  
+  //! Face ids and its associated nodal indices
+  const std::map<unsigned, Eigen::Matrix<int, 4, 1>> face_indices_hexahedron{
+      {0, Eigen::Matrix<int, 4, 1>(0, 1, 5, 4)},
+      {1, Eigen::Matrix<int, 4, 1>(5, 1, 2, 0)},
+      {2, Eigen::Matrix<int, 4, 1>(7, 6, 2, 3)},
+      {3, Eigen::Matrix<int, 4, 1>(0, 4, 7, 3)},
+      {4, Eigen::Matrix<int, 4, 1>(1, 0, 3, 2)},
+      {5, Eigen::Matrix<int, 4, 1>(4, 5, 6, 7)}};
+
+  try {
+    return face_indices_hexahedron.at(face_id);
+  } catch (std::exception& exception) {
+    console_->error("{} #{}: {}\n", __FILE__, __LINE__, exception.what());
+  }
+}
+
+//! Return indices of a face of the element
+//! 20-noded hexahedron
+template <>
+inline Eigen::MatrixXi
+    mpm::HexahedronElement<3, 20>::face_indices(unsigned face_id) const {
   
+  //! Face ids and its associated nodal indices
+  // clang-format off
+  const std::map<unsigned, Eigen::Matrix<int, 8, 1>> face_indices_hexahedron{
+      {0, (Eigen::Matrix<int, 8, 1>() << 0, 1, 5, 4,  8, 12, 16, 10).finished()},
+      {1, (Eigen::Matrix<int, 8, 1>() << 5, 1, 2, 0, 12, 11, 14, 18).finished()},
+      {2, (Eigen::Matrix<int, 8, 1>() << 7, 6, 2, 3, 19, 14, 13, 15).finished()},
+      {3, (Eigen::Matrix<int, 8, 1>() << 0, 4, 7, 3, 10, 17, 15,  9).finished()},
+      {4, (Eigen::Matrix<int, 8, 1>() << 1, 0, 3, 2,  8,  9, 13, 11).finished()},
+      {5, (Eigen::Matrix<int, 8, 1>() << 4, 5, 6, 7, 16, 18, 19, 17).finished()}};
+  //clang-format on
+  try {
+    return face_indices_hexahedron.at(face_id);
+  } catch (std::exception& exception) {
```

Comment:
I wouldn't do a try block here, because if you catch  an exception here, you are not returning anything to hope for it to continue, which will create more problems than handling this exception

Reply:
I agree since `map` its own exception.